src/otx/core/data/dataset/keypoint_detection.py

```diff
@@ -0,0 +1,134 @@
+            )
+        else:
+            self.label_info = NullLabelInfo()
+
+    def _get_single_bbox_dataset(self, dm_subset: DatasetSubset) -> Dataset:
+        """Method for splitting dataset itmes into multiple items for each bbox/keypoint."""
+        dm_items = []
+        for item in dm_subset:
+            new_items = defaultdict(list)
+            for ann in item.annotations:
+                if isinstance(ann, (Bbox, Points)):
+                    new_items[ann.id].append(ann)
+            for ann_id, anns in new_items.items():
+                available_types = []
+                for ann in anns:
+                    if isinstance(ann, Bbox) and (ann.w <= 0 or ann.h <= 0):
+                        continue
+                    if isinstance(ann, Points) and max(ann.points) <= 0:
+                        continue
+                    available_types.append(ann.type)
+                if available_types != [AnnotationType.points, AnnotationType.bbox]:
+                    continue
+                dm_items.append(item.wrap(id=item.id + "_" + str(ann_id), annotations=anns))
+        return Dataset.from_iterable(dm_items, categories=self.dm_subset.categories())
+
+    def _get_item_impl(self, index: int) -> KeypointDetDataEntity | None:
+        item = self.dm_subset[index]
+        img = item.media_as(Image)
+        ignored_labels: list[int] = []  # This should be assigned form item
+        img_data, img_shape = self._get_img_data_and_shape(img)
+
+        bbox_anns = [ann for ann in item.annotations if isinstance(ann, Bbox)]
+        bboxes = (
+            np.stack([ann.points for ann in bbox_anns], axis=0).astype(np.float32)
+            if len(bbox_anns) > 0
+            else np.zeros((0, 4), dtype=np.float32)
+        )
+
+        # keypoints in shape [1, K, 2] and keypoints_visible in [1, K]
+        keypoint_anns = [ann for ann in item.annotations if isinstance(ann, Points)]
+        keypoints = (
+            np.stack([ann.points for ann in keypoint_anns], axis=0).astype(np.float32)
+            if len(keypoint_anns) > 0
+            else np.zeros((0, 4), dtype=np.float32)
+        ).reshape(-1, 2)
+        keypoints_visible = np.minimum(1, keypoints)[..., 0]
+
+        entity = KeypointDetDataEntity(
+            image=img_data,
+            img_info=ImageInfo(
+                img_idx=index,
+                img_shape=img_shape,
+                ori_shape=img_shape,
+                image_color_channel=self.image_color_channel,
+                ignored_labels=ignored_labels,
+            ),
+            bboxes=tv_tensors.BoundingBoxes(
+                bboxes,
+                format=tv_tensors.BoundingBoxFormat.XYXY,
+                canvas_size=img_shape,
+            ),
+            keypoints=tv_tensors.TVTensor(keypoints),
+            keypoints_visible=tv_tensors.TVTensor(keypoints_visible),
+            labels=torch.as_tensor([ann.label for ann in bbox_anns]),
+            keypoint_x_labels=tv_tensors.TVTensor([]),
```

Comment:
Do they always empty here? On which stage do we assign some values to these parameters?